### PR TITLE
Cleanup docker layers in mbzirc_sim Dockerfile

### PR DESCRIPTION
**Pull request by j-rivero**
Mostly clean artifacts created on the runs of every layer. I left a couple of question on the same code before merging

### CODE REVIEW

File: docker/mbzirc_sim/Dockerfile [hunk 1]
```diff
@@ -115,14 +130,18 @@ ENV IGNITION_VERSION fortress
...
+  && sudo apt-get clean -qq
 
-RUN /bin/bash -c 'source /opt/ros/galactic/setup.bash && colcon build --merge-install'
+RUN /bin/bash -c 'source /opt/ros/galactic/setup.bash \
+  && colcon build --merge-install' \
+  && rm -fr build/ log/
 RUN /bin/sh -c 'echo ". /opt/ros/galactic/setup.bash" >> ~/.bashrc' \
  && /bin/sh -c 'echo ". ~/mbzirc_ws/install/setup.sh" >> ~/.bashrc'
 
-# Create a directory required by cloudsim
```
Comment (j-rivero) [inline docker/mbzirc_sim/Dockerfile:124]: Are we planning on using cloudsim at some point?

File: docker/mbzirc_sim/Dockerfile [hunk 2]
```diff
@@ -115,14 +130,18 @@ ENV IGNITION_VERSION fortress
...
-RUN mkdir -p /tmp/ign
+# Leave apt-get ready for users
```
Comment (j-rivero) [inline docker/mbzirc_sim/Dockerfile:143]: Not sure how would be the interaction of the users with this docker software. 
 * Do they need to install anything? If not we can remove this.
 * Should the run script be run automatically when users run the docker layer? if so, we need enable lines below.

Comment (j-rivero) [reply docker/mbzirc_sim/Dockerfile]: ack, afeb04c .

File: docker/mbzirc_sim/Dockerfile [hunk 3]
```diff
@@ -99,12 +105,6 @@ RUN mkdir ~/workspaces ~/other
 # When running a container start in the developer's home folder
 WORKDIR /home/$USERNAME
 
-# clone mbzirc sim
-RUN mkdir -p mbzirc_ws/src \
- && cd mbzirc_ws/src \
- && git config --global http.postBuffer 1048576000 \
- && git clone https://github.com/osrf/mbzirc
```
Comment (iche033) [inline docker/mbzirc_sim/Dockerfile:106]: we shouldn't remove this in this PR right? Or maybe #18 should be target at this branch

Comment (j-rivero) [reply docker/mbzirc_sim/Dockerfile:106]: Sorry the intention was to open two PRs against main but I failed to do it properly. Let's do this and I reopen the next one.

Comment (iche033) [reply docker/mbzirc_sim/Dockerfile]: ok I reverted this line otherwise the next cmd for cd'ing into the directory will fail. a8203c3

**Review by iche033 — APPROVED** [review]

**Pull request merged**